seb_openedx/user_banning.py

```diff
@@ -0,0 +1,93 @@
+    last_modified_time = datetime.now()
+    _get_back_end().ban_user(username, course_key, last_modified_time, banned_by)
+
+
+def unban_user(username, course_key, banned_by):
+    """ exposed function to check if user is already banned """
+    last_modified_time = datetime.now()
+    _get_back_end().unban_user(username, course_key, last_modified_time, banned_by)
+
+
+@six.add_metaclass(abc.ABCMeta)
+class BannedUsersBackend(object):
+    """ Abstract BannedUserBackend class """
+
+    @abc.abstractmethod
+    def is_user_banned(self, username, course_key):
+        """ is_user_banned abstract method """
+        pass
+
+    @abc.abstractmethod
+    def ban_user(self, username, course_key, last_modified_time, banned_by):
+        """ is_user_banned abstract method """
+        pass
+
+    @abc.abstractmethod
+    def unban_user(self, username, course_key, last_modified_time, unbanned_by):
+        """ is_user_banned abstract method """
+        pass
+
+
+class DatabaseBannedUsersBackend(BannedUsersBackend):
+    """ Database backend """
+    def is_user_banned(self, username, course_key):
+        """ is_user_banned implementation """
+        return len(ForbiddenCourseAccess.objects.filter(username=username, course_id=course_key))  # pylint: disable=no-member
+
+    def ban_user(self, username, course_key, last_modified_time, banned_by):
+        """ ban_user implementation """
+        forbidden_access, created = ForbiddenCourseAccess.objects.get_or_create(  # pylint: disable=unused-variable
+            username=username,
+            course_id=course_key,
+        )
+        forbidden_access.access_blocked = True
+        forbidden_access.last_modified_time = last_modified_time
+        forbidden_access.last_modified_by_username = banned_by
+        forbidden_access.save()
+
+    def unban_user(self, username, course_key, last_modified_time, unbanned_by):
+        """ ban_user implementation """
+        forbidden_access = ForbiddenCourseAccess.objects.filter(username=username, course_id=course_key).first()
+        forbidden_access.access_blocked = False
+        forbidden_access.last_modified_time = last_modified_time
+        forbidden_access.last_modified_by_username = unbanned_by
+        forbidden_access.save()
+
+
+class UserprofileBannedUsersBackend(BannedUsersBackend):
+    """ Userprofile backend """
+    def is_user_banned(self, username, course_key):
+        """ is_user_banned implementation """
+        raise NotImplementedError()
+
+    def ban_user(self, username, course_key, last_modified_time, banned_by):
+        """ ban_user implementation """
+        raise NotImplementedError()
+
+    def unban_user(self, username, course_key, last_modified_time, unbanned_by):
+        """ ban_user implementation """
+        raise NotImplementedError()
+
+
+def _get_back_end():
+    """ get back end util """
+    if getattr(settings, 'SEB_USER_BANNING_BACKEND', 'database') == 'database':
```

Comment:
to be more in line with django, the name of the backend should be 'seb_openedx.user_banning.DatabaseBannedUsersBackend' or 'DatabaseBannedUsersBackend' if we are already asuming the backend lives in this module and here the import would require no special case scenario for the `'database'` string.

Reply:
Ok